std/range/package.d

```diff
@@ -9740,79 +9742,166 @@ struct Bitwise(R)
         in
         {
             assert(n >= 0);
+            // TODO slice.d 2527
+
+            /*
+               If it does not have the length property, it means that R is
+               an infinite range
+             */
             static if (hasLength!R)
             {
-                import core.exception : RangeError;
-                import std.exception : enforce;
-
-                enforce(n < len, new RangeError);
+                assert(n < len,
+                       __PRETTY_FUNCTION__ ~ ": Index out of bounds");
             }
         }
         body
         {
             if (maskPos > n)
             {
-                UER indexMask = mask >> n;
-                return cast(bool)(accumulator & indexMask);
+                return (parent.front & mask(maskPos - n)) != 0;
             }
 
             n -= maskPos;
 
             /*
-               We need to subtract 1 from the element index because we have
-               the Bitwise.accumulator holding the first element of the range
-               and now parent[0] holds the next element after accumulator.
-
-               If n is less than ER.sizeof, but we have already consumed from
-               the accumulator more than n bits, then elemIndex will be -1,
-               because we want the remainder bits from parent[0]. Because of
-               this, we need to check if elemIndex is negative and set it to 0.
+               By truncating n with maskPos bits we have skipped the remaining
+               maskPos bits in parent[0], so we need to add 1 to elemIndex.
              */
-            long elemIndex = n / bitsNum - 1;
-            if (elemIndex < 0)
-            {
-                elemIndex = 0;
-            }
+            size_t elemIndex = n / bitsNum + 1;
```

Comment:
Do you think that the following construct would be faster?
`elemIndex = n >> (bitsNum.bsf + 1) + 1;`